### PR TITLE
Ensure subscribed to minimum subnets

### PR DESCRIPTION
**Pull request by courtneyeh**
#6580 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/subnets/ValidatorBasedStableSubnetSubscriber.java [hunk 1]
```diff
@@ -93,6 +97,7 @@ private Set<SubnetSubscription> adjustNumberOfSubscriptionsToNumberOfValidators(
     final int randomSubnetsPerValidator = ValidatorConstants.RANDOM_SUBNETS_PER_VALIDATOR;
     int totalNumberOfSubscriptions =
         min(ATTESTATION_SUBNET_COUNT, randomSubnetsPerValidator * validatorCount);
+    totalNumberOfSubscriptions = max(totalNumberOfSubscriptions, minimumSubnetSubscriptions);
```
Comment (rolfyone) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/subnets/ValidatorBasedStableSubnetSubscriber.java]: should really make totalNumberOfSubscriptions final, and clean this bit up...
I'd probably look at making your new definition of `totalNumberOfSubscriptions` be the final, and clean up the interim calculation....
```
final int n = min(ATTESTATION_SUBNET_COUNT, ValidatorConstants.RANDOM_SUBNETS_PER_VALIDATOR * validatorCount);
final int totalNumberOfSubscriptions = max(n, minimumSubnetSubscriptions);
```
I wouldn't call the variable `n` potentially :) but im not sure what a good name would be :)

**Review by ajsutton — COMMENTED** [review]
If I set `Xp2p-minimum-subnet-subscriptions: 256` I get:

```
2022-12-19 06:47:56.505 | SlotEventsChannel-11      | WARN  | teku-status-log                | Spec failed for event 'interface tech.pegasys.teku.ethereum.events.SlotEventsChannel.onSlot' in handler 'tech.pegasys.teku.validator.coordinator.ActiveValidatorTracker': java.lang.IllegalArgumentException: bound must be positive
java.lang.IllegalArgumentException: bound must be positive
	at java.util.Random.nextInt(Random.java:322) ~[?:?]
	at tech.pegasys.teku.networking.eth2.gossip.subnets.ValidatorBasedStableSubnetSubscriber.getRandomSetElement(ValidatorBasedStableSubnetSubscriber.java:157) ~[classes/:?]
	at tech.pegasys.teku.networking.eth2.gossip.subnets.ValidatorBasedStableSubnetSubscriber.getRandomAvailableSubnetId(ValidatorBasedStableSubnetSubscriber.java:153) ~[classes/:?]
	at tech.pegasys.teku.networking.eth2.gossip.subnets.ValidatorBasedStableSubnetSubscriber.subscribeToNewRandomSubnet(ValidatorBasedStableSubnetSubscriber.java:130) ~[classes/:?]
	at tech.pegasys.teku.networking.eth2.gossip.subnets.ValidatorBasedStableSubnetSubscriber.adjustNumberOfSubscriptionsToNumberOfValidators(ValidatorBasedStableSubnetSubscriber.java:114) ~[classes/:?]
	at tech.pegasys.teku.networking.eth2.gossip.subnets.ValidatorBasedStableSubnetSubscriber.onSlot(ValidatorBasedStableSubnetSubscriber.java:83) ~[classes/:?]
	at tech.pegasys.teku.validator.coordinator.ActiveValidatorTracker.onSlot(ActiveValidatorTracker.java:57) ~[classes/:?]
	at jdk.internal.reflect.GeneratedMethodAccessor3.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at tech.pegasys.teku.infrastructure.events.DirectEventDeliverer.executeMethod(DirectEventDeliverer.java:74) ~[classes/:?]
	at tech.pegasys.teku.infrastructure.events.DirectEventDeliverer.deliverTo(DirectEventDeliverer.java:54) ~[classes/:?]
	at tech.pegasys.teku.infrastructure.events.AsyncEventDeliverer.lambda$deliverTo$0(AsyncEventDeliverer.java:66) ~[classes/:?]
	at tech.pegasys.teku.infrastructure.events.AsyncEventDeliverer$QueueReader.deliverNextEvent(AsyncEventDeliverer.java:125) ~[classes/:?]
	at tech.pegasys.teku.infrastructure.events.AsyncEventDeliverer$QueueReader.run(AsyncEventDeliverer.java:116) ~[classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

Would probably make a good test case to add. :)

**Review by rolfyone — APPROVED** [review]
LGTM

**Review by ajsutton — APPROVED** [review]
LGTM.

**Pull request merged**